src/autora/state/wrapper.py

```diff
@@ -85,3 +88,164 @@ def experimentalist(params):
+
+        >>> experimentalist_on_state(S(conditions=[1,2,3,4]))
+        S(conditions=[11, 12, 13, 14])
+
+        >>> experimentalist_on_state(S(conditions=[101,102,103,104]))
+        S(conditions=[111, 112, 113, 114])
+
+        >>> from autora.variable import VariableCollection, Variable
+        >>> from sklearn.base import BaseEstimator
+        >>> from sklearn.linear_model import LinearRegression
+
+        We can use other states with different fields:
+        >>> @dataclass(frozen=True)
+        ... class T(State):
+        ...     variables: VariableCollection  # field(metadata={"delta":... }) omitted ∴ immutable
+        ...     experiment_data: pd.DataFrame = field(metadata={"delta": "extend"})
+        ...     model: Optional[BaseEstimator] = field(metadata={"delta": "replace"}, default=None)
+
+        >>> t = T(
+        ...     variables=VariableCollection(independent_variables=[Variable("x")],
+        ...                                  dependent_variables=[Variable("y")]),
+        ...     experiment_data=pd.DataFrame({"x": [0,1,2,3,4], "y": [2,3,4,5,6]})
+        ... )
+
+        And define other functions:
+        >>> def theorist(x: pd.DataFrame, v: VariableCollection, **kwargs):
+        ...     ivs = [v.name for v in v.independent_variables]
+        ...     dvs = [v.name for v in v.dependent_variables]
+        ...     X, y = x[ivs], x[dvs]
+        ...     new_model = LinearRegression(fit_intercept=True).set_params(**kwargs).fit(X, y)
+        ...     return new_model
+
+        Again, we use the wrapper:
+        >>> theorist_on_state = wrap_to_use_state_general(
+        ...                                         theorist,
+        ...                                         {'x':'experiment_data', 'v':'variables'},
+        ...                                          'model')
+
+        >>> t_prime = theorist_on_state(t)
+        >>> t_prime.model.coef_, t_prime.model.intercept_
+        (array([[1.]]), array([2.]))
+
+
+        Other arguments are still supported by the inner function can also be passed
+        (if and only if the inner function allows for and handles `**kwargs` arguments alongside
+        the values from the state).
+        >>> theorist_on_state(t, fit_intercept=False).model.intercept_
+        0.0
+
+        Any parameters not provided by the state can still be provided by to the wrapped function
+        >>> def experimentalist(x, offset):
+        ...     new_conditions = [c + offset for c in x]
+        ...     return new_conditions
+
+        >>> experimentalist_on_state_ = wrap_to_use_state_general(
+        ...                                                 f=experimentalist,
+        ...                                                 input_state_mapping={'x':'conditions'},
+        ...                                                 output_field_name='conditions')
+
+
+        ... then it need to be passed.
```

Comment:
need this? not quite sure how to read as a continuation of previous sentence (which seems comprehensive)